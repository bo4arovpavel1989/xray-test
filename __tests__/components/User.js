import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import { User } from '../../src/components/Test/User';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Menu component', () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn()
  };
  const component = mount(
    <MemoryRouter initialEntries={[ { key: 'testKey' } ]}>
      <User { ...props }/>
    </MemoryRouter>
  );

  it('should render correct', () => {
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
