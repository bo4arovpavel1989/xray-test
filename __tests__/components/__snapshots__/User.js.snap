// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should render correct 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <User
      handleChange={[MockFunction]}
      handleSubmit={[MockFunction]}
    >
      <div>
        <div
          className="paranja"
        />
        <div
          className="userLoginArea"
        >
          <form
            className="userLogin"
            onSubmit={[MockFunction]}
          >
            <div>
              <span>
                <label>
                  Ваше имя:  
                  <input
                    id="user"
                    onChange={[MockFunction]}
                    required={true}
                    type="text"
                  />
                </label>
              </span>
              <span>
                <input
                  type="submit"
                  value="Начать тест"
                />
              </span>
            </div>
          </form>
        </div>
      </div>
    </User>
  </Router>
</MemoryRouter>
`;
